File: competitive-st/Day Month.py
Day = int(input("Day : "))
Month = int(input("Month : "))

list_week = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday",
             "Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]

if Day < 1:
    print("Error")
elif Month in [1,3,5,7,8,10,12] and Day > 31:
    print("Error")
elif Month in [4,6,9,11] and Day > 30:
    print("Error")
elif Month == 2 and Day > 29:
    print("Error")
else:
    Day1 = Day % 7
    if Month in [9,12]:
        print(list_week[Day1 - 1])

    elif Month in [1,4,7]:
        print(list_week[Day1])

    elif Month in [10]:
        print(list_week[Day1 + 1])

    elif Month in [5]:
        print(list_week[Day1 + 2])

    elif Month in [2,8]:
        print(list_week[Day1 + 3])

    elif Month in [3,11]:
        print(list_week[Day1 + 4])

    elif Month in [6]:
        print(list_week[Day1 + 5])
    else:
        print("Error")

File: competitive-st/C++ Code/DayMonth.cpp
#include <iostream>
#include <vector>
using namespace std;
int checkDate(int day , int month){
    int date = day % 7;
    if(day < 1){
        return 200;
    }else if((month == 1 ||month == 3 ||month == 5 ||month == 7 ||month == 8 ||month == 10 ||month == 12) && day > 31){
        return 200;
    }else if((month == 4 ||month == 6 ||month == 9 ||month == 11) && day > 30){
        return 200;
    }else if(month == 2 && day > 29){
        return 200;
    }else{
        if (month == 9 || month == 12){
            return (date - 1);
        }else if (month == 1 || month == 4 || month == 7){
            return date;
        }else if (month == 10){
            return (date + 1);
        }else if (month == 5){
            return (date + 2);
        }else if (month == 2 || month == 8){
            return (date + 3);
        }else if (month == 3 || month == 11){
            return (date + 4);
        }else if (month == 6){
            return (date + 5);
        }else{
            return 200;
        }
    }
}
int main(){
    int day;
    int month;
    vector<string> dates = {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday","Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};
    cout << "Input Day = ";
    cin >> day;
    cout << "Input Month = ";
    cin >> month;
    int indexReturn = checkDate(day,month);
    if (indexReturn == 200){
        cout << "Error";
    }else{
        cout << dates[indexReturn];
    }
    return 0;
}
